### PR TITLE
Additional item endpoints ar

### PR DESCRIPTION
**Pull request by reid-andrew**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling - no new features
### Why is this change required? What problem does it solve?
- Refactors existing items endpoints & adds additional endpoints for full CRUD of Items.
- Previous approach was creating problems when attempting to run POSTs & PUTs
### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
- There was a bit of a learning curve attempting to implement the methods from Flask_Restful. Lots of trial & error & research to resolve.

### CODE REVIEW

File: app/controllers/items.py [hunk 1]
```diff
@@ -1,21 +1,136 @@
...
+    'charity': fields.String,
+    'image': fields.String
+}
+
+item_list_fields = {
+    'count': fields.Integer,
+    'items': fields.List(fields.Nested(item_fields))
+}
+
+item_post_parser = reqparse.RequestParser()
```
Comment (reid-andrew) [inline app/controllers/items.py:25]: Note: in the add argument statements below we've set each of these fields as required. I don't believe there are any situations where we would be OK with any of them being blank, but may want to review that with FE team.

File: app/controllers/items.py [hunk 2]
```diff
@@ -1,21 +1,136 @@
...
+        if item_id:
+            item = Item.query.filter_by(id=item_id).first()
+            return marshal(item, item_fields)
+        else:
+            items = Item.query.all()
+            return marshal({
+                'count': len(items),
+                'items': [marshal(i, item_fields) for i in items]
+            }, item_list_fields)
+
```
Comment (reid-andrew) [inline app/controllers/items.py:94]: Defining both GET requests (Index & Show) in this one method, based on whether /items or /items/id is hit. A bit different from how we structured things in Rails projects, but this appears consistent with what I'm seeing elsewhere.

File: app/controllers/items.py [hunk 3]
```diff
@@ -1,21 +1,136 @@
...
+        db.session.commit()
+        return item
+
+    @marshal_with(item_fields)
+    def delete(self, item_id=None):
+        item = Item.query.get(item_id)
+
+        db.session.delete(item)
+        db.session.commit()
```
Comment (StephanieFriend) [inline app/controllers/items.py:134]: I really like how this is laid out. Very readable and clean!

File: app/models/item.py [hunk 4]
```diff
@@ -9,23 +9,9 @@ class Item(db.Model):
...
-
-    def __init__(self, title, description, price, donor, status):
-        self.title = title
-        self.description = description
-        self.price = price
-        self.donor = donor
-        self.status = status
+    category = db.Column(db.String())
+    charity = db.Column(db.String())
+    image = db.Column(db.String())
```
Comment (StephanieFriend) [inline app/models/item.py:14]: Thanks for adding these per the FE request!

**Review by StephanieFriend — APPROVED** [review]
This all looks great! Hopefully we'll be able to add some tests soon. :)

File: app/controllers/items.py (hunk 2 above)
Comment (kmcgrevey) [reply app/controllers/items.py:94]: Noted that conditional as well -- interesting convention

File: app/__init__.py [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
 db = SQLAlchemy(app)
 
+api = Api(app)
 
 from app.models.item import Item
-from app.controllers.items import ItemRoutes
+from app.controllers.items import ItemResources
 from app.controllers.welcome import WelcomeRoutes
 
+api.add_resource(ItemResources, '/items', '/items/<int:item_id>')
```
Comment (kmcgrevey) [inline app/__init__.py:18]: endpoints!

File: app/controllers/items.py [hunk 6]
```diff
@@ -1,21 +1,136 @@
 from app import app, db
 from flask import Flask, request, jsonify
 from flask_sqlalchemy import SQLAlchemy
+from flask_restful import Resource, reqparse, request
+from flask_restful import fields, marshal_with, marshal
```
Comment (kmcgrevey) [inline app/controllers/items.py:5]: Looks like `Flask_RESTful` extension cleaned up some syntax.  Moved the url endpoints to the `__init__.py` in `app` folder

**Review by kmcgrevey — APPROVED** [review]

**Pull request merged**